### PR TITLE
feat: Setup feature dag to support matrix values for scripture

### PR DESCRIPTION
**Pull request by vinnyjth**
### About

Allows Matrix Attribute Values (used on Newspring) to work properly with the features system. This code stops empty matrix attributes from showing up, and properly builds scripture features from a valid matrix attribute value. 

image

Before
Screen Shot 2021-09-28 at 3 36 09 PM

After
Screen Shot 2021-09-28 at 4 41 57 PM

### Testing

I don't have a good way to test this in core :/ Newspring has a lot of matrix-value attributes though, so I can walk you through pulling down Newspring and running the code if you're interested. I did write a test for this to help verify functionality. 

### Screenshots
See above :)

### CODE REVIEW

File: dags/rock/rock_features.py [hunk 1]
```diff
@@ -165,23 +167,62 @@ def check_postgres_features(self, rock_content_item_features, content_item_id):
...
 
         if scriptures:
-            for scripture in scriptures.split(","):
-                key_value_features.append({"key": "scripture", "value": scripture})
+            scripture_field_type = safeget(
+                content, "Attributes", "Scriptures", "FieldTypeId"
+            )
+
+            # If the attribute is a matrix.
+            if scripture_field_type == self.get_attribute_matrix_field_type_id():
```
Comment (tomthornton) [inline dags/rock/rock_features.py:199]: 👍

File: tests/test_rock_features.py [hunk 2]
```diff
@@ -67,7 +67,9 @@ def mock_get(config, deserialize_json=True, default_var=None):
 
     with vcr.use_cassette("tests/cassettes/features/content_items.yaml"):
         content_item.run_fetch_and_save_content_items()
-    with vcr.use_cassette("tests/cassettes/features/initial_features.yaml"):
+    with vcr.use_cassette(
+        "tests/cassettes/features/initial_features.yaml", record_mode="new_episodes"
```
Comment (tomthornton) [inline tests/test_rock_features.py:71]: What does "new_episodes" do?

File: tests/test_rock_features.py [hunk 3]
```diff
@@ -142,3 +150,78 @@ def mock_get(config, deserialize_json=True, default_var=None):
                 i += 1
 
     conn.close()
+
+
+def test_matrix_value_scriptures(monkeypatch):
```
Comment (tomthornton) [inline tests/test_rock_features.py:155]: Great test!

**Review by tomthornton — APPROVED** [review]
Looks good to me! Pulled it down and the tests ran fine.

File: tests/test_rock_features.py (hunk 2 above)
Comment (vinnyjth) [reply tests/test_rock_features.py:71]: It allows new snapshots to be added, but does not allow existing snapshots to be changed.

**Pull request merged**